apm-agent-plugins/apm-scala-concurrent-plugin/src/main/scala/co/elastic/apm/agent/concurrent/FutureInstrumentation.scala

```diff
@@ -0,0 +1,41 @@
+package co.elastic.apm.agent.concurrent
+
+import java.util
+
+import co.elastic.apm.agent.bci.ElasticApmInstrumentation
+import net.bytebuddy.asm.Advice
+import net.bytebuddy.description.`type`.TypeDescription
+import net.bytebuddy.description.method.MethodDescription
+import net.bytebuddy.matcher.ElementMatcher
+import net.bytebuddy.matcher.ElementMatchers._
+
+import scala.collection.JavaConverters._
+import scala.util.Try
+
+class FutureInstrumentation extends ElasticApmInstrumentation {
+
+  override def getTypeMatcher: ElementMatcher[_ >: TypeDescription] =
+    hasSuperType[TypeDescription](named("scala.concurrent.Future"))
+      .or(hasSuperType(named("scala.concurrent.impl.Promise")))
+      .or(hasSuperType(named("scala.concurrent.impl.Promise$Transformation")))
+      .or(hasSuperType(named("scala.concurrent.Future$")))
+
+  override def getMethodMatcher: ElementMatcher[_ >: MethodDescription] =
+    named[MethodDescription]("onComplete").and(returns(classOf[Unit])).and(takesArguments(classOf[Try[_] => _]))
```

Comment:
The type matchers should not refer to classes as that would cause loading these classes. Instead, use the `named` matcher, like `named("scala.util.Try")`